apps/vanilla/src/pages/exercise2/scripts/types.ts

```diff
@@ -0,0 +1,53 @@
+/** ISubscriber - interface for subscribers. */
+export interface ISubscriber<T>{
+
+	/** Function update subscriber. */
+	update(data: T): void;
+}
+
+/** Data about who is walking. */
+export interface ICurrentPlayerMoveData {
+
+	/** Current player's index. */
+	readonly currentPlayerId: number;
+
+	/** Next player's index. */
+	readonly nextPlayerId: number;
+}
+
+/** Data about who is walking. */
+export interface IMoveData extends ICurrentPlayerMoveData {
+
+	/** Number of the side of the dice that has fallen. */
+	readonly diceSide: number;
+}
+
+/** Data to display the result. */
+export interface IDisplayStatusData{
+
+	/** Player status. */
+	readonly status: TDisplayStatus[];
```

Comment:
Don't forget about readonly for arrays:

```suggestion
	readonly status: readonly TDisplayStatus[];
```

The only time we don't recommend using `readonly` for an array is for the result returned from `function`/methods.